datalad/core/distributed/push.py

```diff
@@ -123,16 +116,27 @@ class Push(Interface):
             data or changes for those paths are considered for a push.""",
             nargs='*',
             constraints=EnsureStr() | EnsureNone()),
+        transfer_data=Parameter(
+            args=("--transfer-data",),
+            doc="""what to do with data (annex'ed) data. 'anything' would cause
+            transfer of all annexed content, 'nothing' would avoid call to
+            `git annex copy` altogether. 'auto' would use 'git annex copy' with
+            '--auto' thus transferring only data which would satisfy "wanted"
+            or "numcopies" settings for the remote (thus "nothing" otherwise).
+            'auto-if-wanted' would enable '--auto' mode only if there is a 
+            "wanted" setting for the remote, and transfer 'anything' otherwise.
+            Note: 'anything' and 'nothing' are "wanted" expressions understood
+            by git-annex.""",
```

Comment:
What purpose has the "note"? These values are not actually passed on to git-annex, hence it seems the note merely points out an inconsequential similarity.

Reply:
It avoids a possible user questions:
- why not just 'all' and 'nothing'? (or any other pair of words)
- why not just 'auto', 'none', 'all' as `publish` has it?

I did not want to unnecessarily expand it here, but  overall docstring above or other docs could be extended to state e.g. that "instead of specifying explicitly `--transfer-data=nothing` to `push`, you can configure the specific remote via `git annex wanted REMOTE nothing` to make no data pushed to that particular remote by default."

Reply:
but also it probably asks for RFing `publish` to use those values, which could be done for 0.13, possibly just introducing equivalences of `none` == `nothing`, and `all` == `anything`; while issuing deprecation warnings whenever someone uses "old style" values. With `publish` RFing later on (I could attempt it on top of this PR to verify if my assumptions are correct) to use `push` - it would probably be just a matter of changing default from `auto-if-wanted` to `auto` and be done (for the `push` aspect altogether.  Will need to RF also just to call `create-sibling` first ;))

Reply:
I also felt this note was out of place.

Reply:
ok, I will remove it then.